### DISCUSSION TITLE
How to inject a MemoryAllocator?

### DISCUSSION THREAD

**Discussion by antonfirsov** [SixLabors/ImageSharp.Web#152] [Q&A] [answered]
Coming from https://github.com/SixLabors/ImageSharp/discussions/1590, I had a look through the codebase to understand how `MemoryAllocator` is defaulted and what is the standard way to change the defaults.

I found several overloads for [`SetMemoryAllocator`](https://github.com/SixLabors/ImageSharp.Web/blob/2c5961e6161011b40ea7685da54017fcd92b7b87/src/ImageSharp.Web/DependencyInjection/ImageSharpBuilderExtensions.cs#L57-L62), but I wasn't able to find code that dispatches the allocator as a service.

I found a `Configuration` instance in `ImageSharpMiddlewareOptions`, I guess that's the point where it is plugged. Are the `SetMemoryAllocator` extensions obsolete then?

**Comment by JimBobSquarePants** [comment, marked as answer]
Yeah, those methods should be obsoleted and/or removed. They don’t do anything and we’re left  it he API by accident. 

Editing the configuration instance should be the way we choose.
